### PR TITLE
#288: fixed integer mapping for BigQueryQueryRewriter

### LINKED ISSUE #288: Big Query: wrong ColumnMapping for BigInt
In BigQueryQueryRewriter:
java datatype int is too small to transform BigQuery BigInt to Exasol Integer.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #288

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java [hunk 1]
```diff
@@ -143,26 +148,23 @@ private void appendDate(final StringBuilder builder, final ResultSet resultSet,
...
+        final Matcher matcher = DATE_PATTERN.matcher(dateToCast);
+        if (matcher.matches()) {
+            final int year = Integer.parseInt(matcher.group(1));
+            final int month = Integer.parseInt(matcher.group(2));
+            final int day = Integer.parseInt(matcher.group(3));
+            return String.format("%02d.%02d.%04d", day, month, year);
         } else {
-            return null;
+            throw new IllegalArgumentException(
+                    "Date does not match required format: YYYY-[M]M-[D]D. Actual " + dateToCast);
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java]:
```suggestion
                    "Date does not match required format: YYYY-[M]M-[D]D. Actual value was: " + dateToCast);
```

File: doc/dialects/bigquery.md [hunk 2]
```diff
@@ -66,15 +70,43 @@ Below you see how a Big Query Virtual Schema is created. Please note that you ha
...
-    CONNECTION_NAME = 'BIGQUERY_CONNECTION'
+    CONNECTION_NAME = 'BIGQUERY_JDBC_CONNECTION'
     CATALOG_NAME = '<catalog name>'
     SCHEMA_NAME = '<schema name>';
 ```
 
+
+## Data Types Conversion
+
+BigQuery Data Type | Support Status | Converted Exasol Data Type| Known limitations
```
Comment (redcatbear) [inline doc/dialects/bigquery.md]:
```suggestion
BigQuery Data Type | Supported | Converted Exasol Data Type| Known limitations
```

File: doc/dialects/bigquery.md [hunk 3]
```diff
@@ -11,35 +11,39 @@ Download the [Simba JDBC Driver for Google BigQuery](https://cloud.google.com/bi
 1. [Create a bucket in BucketFS](https://docs.exasol.com/administration/on-premise/bucketfs/create_new_bucket_in_bucketfs_service.htm) 
 1. Upload the driver to BucketFS
 
+Hint: Magnitude Simba driver contains a lot of jar files, but you can upload all of them together as an archive (tar.gz, for example).
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/dialects/bigquery.md]: done

File: doc/dialects/bigquery.md [hunk 4]
```diff
@@ -11,35 +11,39 @@ Download the [Simba JDBC Driver for Google BigQuery](https://cloud.google.com/bi
...
+
 ## Installing the Adapter Script
 
 Upload the latest available release of [Virtual Schema JDBC Adapter](https://github.com/exasol/virtual-schemas/releases) to Bucket FS.
 
 Then create a schema to hold the adapter script.
 
 ```sql
-CREATE SCHEMA ADAPTER;
+CREATE SCHEMA TEST_SCHEMA;
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/dialects/bigquery.md]: Renamed to `SCHEMA_FOR_VS_SCRIPT`

File: doc/dialects/bigquery.md [hunk 5]
```diff
@@ -66,15 +70,43 @@ Below you see how a Big Query Virtual Schema is created. Please note that you ha
...
+
+## Data Types Conversion
+
+BigQuery Data Type | Support Status | Converted Exasol Data Type| Known limitations
+-------------------|----------------|---------------------------|--------------------------------------------------
+BOOLEAN            |  ✓             | BOOLEAN                   | 
+BYTES              |  ×             |                           | 
+DATE               |  ✓             | DATE                      | 
+DATETIME           |  ✓             | TIMESTAMP                 | 
+FLOAT              |  ✓             | DOUBLE                    | Expected range for correct mapping: -99999999.99999999 .. 99999999.99999999.
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/dialects/bigquery.md]: Agreed to leave it be

File: doc/dialects/bigquery.md [hunk 6]
```diff
@@ -66,15 +70,43 @@ Below you see how a Big Query Virtual Schema is created. Please note that you ha
...
 ## How to improve performance?
 
 Please be aware that the current implementation of the dialect can only handle result sets with limited size (a few thousand rows).
-If you need to proceed a large amount of data, please, contact our support team and create a support ticket. We can provide another implementation of the dialect the with a speed improvement that is not available officially on our GitHub yet due to the complex installation process.
\ No newline at end of file
+If you need to proceed a large amount of data, please, contact our support team and create a support ticket. We can provide another implementation of the dialect the with a speed improvement that is not available officially on our GitHub yet due to the complex installation process.
+
+## Testing information
+
+Here you can find information about latest dialect testing.
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/dialects/bigquery.md]: done

File: doc/dialects/bigquery.md [hunk 7]
```diff
@@ -66,15 +70,43 @@ Below you see how a Big Query Virtual Schema is created. Please note that you ha
...
+
+If you need to use currently unsupported data types or find a way around known limitations, please, contact our support team and create a support ticket.
+
 ## How to improve performance?
 
 Please be aware that the current implementation of the dialect can only handle result sets with limited size (a few thousand rows).
-If you need to proceed a large amount of data, please, contact our support team and create a support ticket. We can provide another implementation of the dialect the with a speed improvement that is not available officially on our GitHub yet due to the complex installation process.
\ No newline at end of file
+If you need to proceed a large amount of data, please, contact our support team and create a support ticket. We can provide another implementation of the dialect the with a speed improvement that is not available officially on our GitHub yet due to the complex installation process.
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/dialects/bigquery.md]: done

File: doc/dialects/bigquery.md [hunk 8]
```diff
@@ -66,15 +70,43 @@ Below you see how a Big Query Virtual Schema is created. Please note that you ha
...
+FLOAT              |  ✓             | DOUBLE                    | Expected range for correct mapping: -99999999.99999999 .. 99999999.99999999. 
+GEOGRAPHY          |  ✓             | VARCHAR(65535)            |
+INTEGER            |  ✓             | DECIMAL(19,0)             | 
+NUMERIC            |  ✓             | VARCHAR(2000000)          | 
+RECORD/STRUCT      |  ×             |                           | 
+STRING             |  ✓             | VARCHAR(65535)            | 
+TIME               |  ✓             | VARCHAR(16)               | 
+TIMESTAMP          |  ✓             | TIMESTAMP                 | Expected range for correct mapping: 1582-10-15 00:00:01 .. 9999-12-31 23:59:59.9999. JDBC driver maps dates before 1582-10-15 00:00:01 incorrectly.  Example of incorrect mapping: 1582-10-14 22:00:01 -> 1582-10-04 22:00:01
+
+If you need to use currently unsupported data types or find a way around known limitations, please, contact our support team and create a support ticket.
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/dialects/bigquery.md]: fixed

File: src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java]: Fixed

File: doc/dialects/bigquery.md (hunk 2 above)
Comment (AnastasiiaSergienko) [reply doc/dialects/bigquery.md]: fixed

File: src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java [hunk 9]
```diff
@@ -68,6 +68,18 @@ private String getQueryFromStatement(final SqlStatement statement, final Adapter
         return statement.accept(this.dialect.getSqlGenerationVisitor(context));
     }
 
+    private void appendQueryForEmptyTable(final StringBuilder builder, final ResultSetMetaData metaData)
+            throws SQLException {
+        final int columnCounter = metaData.getColumnCount();
+        builder.append("(");
+        final StringJoiner joiner = new StringJoiner(", ");
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java]:
```suggestion
        final StringJoiner joiner = new StringJoiner(", ");
```

**Pull request merged**